src/utils/uid.util.js

```diff
@@ -1,47 +1,32 @@
-const hash = require('object-hash');
-/**
- * Return a random int, used by `utils.getUid()`.
- *
- * @param {Number} min
- * @param {Number} max
- * @return {Number}
- * @api private
- */
-function getRandomInt(min, max) {
-  return Math.floor(Math.random() * (max - min + 1)) + min;
-}
+const { randomBytes } = require('node:crypto');
 
 /**
- * Validates the date(s) and return the object containing the prefix and date.
- *
- * Prefix may contain 'ge, le, lt, gt'.
- *
- * @param {*} dates
+ * Generates a UUID that matches with FHIR UUID data type https://build.fhir.org/datatypes.html#uuid
+ * Regex: urn:uuid:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}
+ * ex: c757873d-ec9a-4326-a141-556f43239520
  */
-let getUid = function (length) {
-  let uid = '';
-  const chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
-  const charsLength = chars.length;
+let getUuid = function () {
+  // Generate 16 random bytes
+  const bytes = randomBytes(16);
 
-  for (let i = 0; i < length; ++i) {
-    uid += chars[getRandomInt(0, charsLength - 1)];
-  }
+  // Set the version to 4 (UUID v4)
+  bytes[6] = (bytes[6] & 0x0f) | 0x40;
 
-  return uid;
-};
+  // Set the variant to 10xx (UUID v4)
+  bytes[8] = (bytes[8] & 0x3f) | 0x80;
 
-/**
- * Make a hash of the object for use as a UUID.
- * TODO Improve this. Stuck this in just because it's more of a uniqueness guarantee than the above 'getUID' function.
- * TODO If we're actually going to generate hashes, we should probably do it in a more secure manner.
- * @param obj
- * @returns {*|*}
- */
-const getUuid = (obj) => {
-  return hash(obj);
+  // Convert bytes to hex and format as UUID v4
+  const uuid = [
+    bytes.toString('hex').slice(0, 8), // 8 characters for the first part
+    bytes.toString('hex').slice(8, 12), // 4 characters for the second part
+    bytes.toString('hex').slice(12, 16), // 4 characters for the third part
+    bytes.toString('hex').slice(16, 20), // 4 characters for the fourth part
+    bytes.toString('hex').slice(20, 32), // 12 characters for the fifth part
+  ].join('-');
+
+  return `${uuid}`;
```

Comment:
Why not just return crypto.randomUUID()?
https://nodejs.org/api/crypto.html#cryptorandomuuidoptions